src/Platform/Microsoft.Testing.Platform/Requests/ITestExecutionFilter.cs

```diff
@@ -1,9 +1,24 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using Microsoft.Testing.Platform.Extensions.Messages;
+
 namespace Microsoft.Testing.Platform.Requests;
 
 /// <summary>
 /// Represents a filter for test execution.
 /// </summary>
-public interface ITestExecutionFilter;
+public interface ITestExecutionFilter
+{
+    /// <summary>
+    /// Gets a value indicating whether the filter should be used for filtering.
+    /// </summary>
+    bool IsEnabled { get; }
+
+    /// <summary>
+    /// Determines whether the specified test node matches the filter criteria.
+    /// </summary>
+    /// <param name="testNode">The test node to check against the filter.</param>
+    /// <returns><c>true</c> if the test node matches the filter; otherwise, <c>false</c>.</returns>
+    bool MatchesFilter(TestNode testNode);
```

Comment:
Same here, I'm not sure if this is better to be Task-returning or not. (I'll discuss with the team)

Reply:
For sync scenarios I think `Task.FromResult(...)` will be used which uses cached tasks internally